pkg/system/phase4_configuring.go

```diff
@@ -1523,3 +1529,142 @@ func (r *Reconciler) ReconcileNamespaceStores(namespaceResources []nb.NamespaceR
 	}
 	return nil
 }
+
+// ReconcileNoobaaDeletionPolicy reconciles the deletion policy of a Noobaa resource
+// and configures the noobaa admission controller appropriately.
+func (r *Reconciler) ReconcileNoobaaDeletionPolicy() error {
+	noobaaAdmissionEnabled := os.Getenv("ENABLE_NOOBAA_ADMISSION")
+	if strings.ToLower(noobaaAdmissionEnabled) != "true" {
+		return nil
+	}
+
+	rule := GetNoobaaCRDeletionAdmissionRule()
+
+	if r.NooBaa.Spec.CleanupPolicy.AllowNoobaaDeletion {
+		return RemoveRuleFromNoobaaAdmissionWebhook(rule)
+	}
+
+	return AddRuleToNoobaaAdmissionWebhook(rule)
+}
+
+// RemoveRuleFromNoobaaAdmissionWebhook removes the rule from the noobaa internal admission webhook
+// if it exists
+func RemoveRuleFromNoobaaAdmissionWebhook(rule *admissionv1.RuleWithOperations) error {
+	wc := &admissionv1.ValidatingWebhookConfiguration{}
+	wc.SetName("admission-validation-webhook")
+
+	// Find the pre-existing admission webhook
+	if _, _, err := util.KubeGet(wc); err != nil {
+		return fmt.Errorf("failed to get webhook: %v", err)
+	}
+
+	// If no rule was removed then return
+	if !removeAdmissionRule(wc, rule) {
+		return nil
+	}
+
+	if !util.KubeUpdate(wc) {
+		return fmt.Errorf("failed to update webhook")
+	}
+
+	return nil
+}
+
+// AddRuleToNoobaaAdmissionWebhook adds the rule to the noobaa admission webhook if it
+// doesn't exists
+func AddRuleToNoobaaAdmissionWebhook(rule *admissionv1.RuleWithOperations) error {
+	wc := &admissionv1.ValidatingWebhookConfiguration{}
+	wc.SetName("admission-validation-webhook")
+
+	// Find the pre-existing admission webhook
+	if _, _, err := util.KubeGet(wc); err != nil {
```

Comment:
@kfir-payne is there a situation where wc exists but it's not used by NooBaa? 
I mean, maybe it's used by other component?

Reply:
I don't thinks so, the `admission-validation-webhook` supposed to be used only by Noobaa

Reply:
I don't think so, `admission-validation-webhook` supposed to be used only by Noobaa